fedimint-server/src/consensus/server.rs

```diff
@@ -454,7 +458,11 @@ impl ConsensusServer {
                 signed_session_outcome = self.request_signed_session_outcome(session_index) => {
                     let pending_accepted_items = self.pending_accepted_items().await;
 
-                    let (processed, unprocessed) = signed_session_outcome.session_outcome.items.split_at(pending_accepted_items.len());
+                    // this panics if we have more accepted items than the signed session outcome
+                    let (processed, unprocessed) = signed_session_outcome
+                        .session_outcome
+                        .items
+                        .split_at(pending_accepted_items.len());
 
                     assert!(processed.iter().eq(pending_accepted_items.iter()));
```

Comment:
I think a message in that assert (that gets shown in the panic if it fails) would also be nice